.github/workflows/integration-tests.yml

```diff
@@ -0,0 +1,93 @@
+name: Integration Tests
```

Comment:
I think it might be better to define all our CI jobs in a single file. That way we'll have better control over their orchestration, like using `.needs` to establish dependencies, define common parameters via `env`, like the Go version.

Reply:
Sure @yurishkuro that makes sense.

Reply:
The down side is that we can only trigger new runs on a per-workflow basis, meaning that all jobs within the workflow will be restarted.

Reply:
yeah, if we cannot rerun individual jobs then it's better to have them as separate workflows.

It's annoying that we'd have to repeat a lot of boilerplate. At some point we might want to consider auto-generating the YAML files from smaller pieces. I originally was looking at [this approach](https://gist.github.com/elijahr/12fe5e13b1e1bbcd81eeaa7c06429e19) of using YAML aliases, but that only works in a single file.

Reply:
I think we can indeed have multiple workflows in the same file, but it would be a huge YAML no matter how much we Don't-Repeat-Ourselves. Let's see what's the final result, and we can then iterate and refactor.

Reply:
Thank you for the review. I am little confused if I should merge unit test and integration tests into one yaml. Using matrix, definitely made the config much cleaner.

Reply:
Also, I think our badger integration tests broke due to some recent commit.